File: 6/b.go
package main

import (
	"fmt"
	"os"
	"slices"
	"strconv"
	"strings"
)

func main() {
	file, _ := os.ReadFile("input.txt")

	lines := strings.Split(string(file), "\n")
	gTime := ""
	for _, time := range slices.Delete(strings.Split(lines[0], " "), 0, 1) {
		if time != "" {
			gTime += time
		}
	}
	gDistance := ""
	for _, distance := range slices.Delete(strings.Split(lines[1], " "), 0, 1) {
		if distance != "" {
			gDistance += distance
		}
	}

	time, _ := strconv.Atoi(gTime)
	distance, _ := strconv.Atoi(gDistance)

	wins := 0
	for j := 1; j < time; j++ {
		if (time-j)*j > distance {
			wins++
		}
	}

	fmt.Println(wins)
}

File: 6/b.ts
// const input = await Bun.file("test.txt").text();
const input = await Bun.file("input.txt").text();

const [time, distance] = input.split("\n").map((d) =>
  parseInt(
    d
      .split(" ")
      .filter((d) => d !== "")
      .slice(1)
      .join("")
  )
);

let c = 0;
for (let j = 1; j < time; j++) {
  const timeRemain = time - j;
  const myDistance = timeRemain * j;
  if (myDistance > distance) {
    c++;
  }
}

console.log(c);
